### PR TITLE
f: add enable_private_gua argument to aws_vpc_ipam resource

### PR DESCRIPTION
**Pull request by JulesPatry**
### Description
Add tier argument to aws_vpc_ipam resource

### Output from Acceptance Testing
```
go test -timeout 100s -run ^TestAccIPAM_enablePrivateGua$ github.com/hashicorp/terraform-provider-aws/internal/service/ec2
ok      github.com/hashicorp/terraform-provider-aws/internal/service/ec2  44.929s
```

### CODE REVIEW

**Comment by jmpatry** [comment]
@drewmullen, it's been a minute! Do you have a chance to take a look at this pull request?

File: internal/service/ec2/ipam_test.go [hunk 1]
```diff
@@ -426,3 +460,16 @@ resource "aws_vpc_ipam" "test" {
...
 `, tier)
 }
+
+func testAccIPAMConfig_enablePrivateGua(enablePrivateGua string) string {
+	return fmt.Sprintf(`
+data "aws_region" "current" {}
+
+resource "aws_vpc_ipam" "test" {
+	operating_regions {
+		region_name = data.aws_region.current.name
```
Comment (drewmullen) [inline internal/service/ec2/ipam_test.go:470]: Gotta fix the spacing in this test file, CI is going to complain once it runs

**Comment by drewmullen** [comment]
@JulesPatry long time no see! 1 small request in the test HCL string but otherwise looks good!

File: .changelog/38838.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```release-note:note
```
Comment (drewmullen) [inline .changelog/38838.txt:1]: Should be “enhancement” I think instead of note

File: internal/service/ec2/ipam_test.go [hunk 3]
```diff
@@ -217,6 +218,39 @@ func TestAccIPAM_tier(t *testing.T) {
...
+			Steps: []resource.TestStep{
+					{
+							Config: testAccIPAMConfig_enablePrivateGua("true"),
+							Check: resource.ComposeTestCheckFunc(
+									testAccCheckIPAMExists(ctx, resourceName, &ipam),
+									resource.TestCheckResourceAttr(resourceName, "enable_private_gua", acctest.CtTrue),
+							),
+					},
+					{
+							Config: testAccIPAMConfig_enablePrivateGua("false"),
```
Comment (drewmullen) [inline internal/service/ec2/ipam_test.go:239]: Probably need to replace “false” with “acctest.CtFalse”

**Review by drewmullen — CHANGES_REQUESTED** [review]
Ok. Nice work. Few comments above

please also remove the changes to go.mod/sum - those are bumped by external processes

**Pull request closed without merging**